client/config/config.go

```diff
@@ -717,8 +717,24 @@ func DefaultConfig() *Config {
 		NoHostUUID:              true,
 		DisableRemoteExec:       false,
 		TemplateConfig: &ClientTemplateConfig{
-			FunctionDenylist: DefaultTemplateFunctionDenylist,
-			DisableSandbox:   false,
+			FunctionDenylist:   DefaultTemplateFunctionDenylist,
+			DisableSandbox:     false,
+			BlockQueryWaitTime: helper.TimeToPtr(5 * time.Minute),         // match Consul default
+			MaxStale:           helper.TimeToPtr(DefaultTemplateMaxStale), // match Consul default
+			Wait: &WaitConfig{
+				Min: helper.TimeToPtr(5 * time.Second),
+				Max: helper.TimeToPtr(4 * time.Minute),
+			},
+			ConsulRetry: &RetryConfig{
+				Attempts:   helper.IntToPtr(0), // unlimited
+				Backoff:    helper.TimeToPtr(250 * time.Millisecond),
+				MaxBackoff: helper.TimeToPtr(1 * time.Minute),
```

Comment:
If we weren't setting these, they'd use the CT default (right?), so setting them here to the default is really more for our own documentation. Do we want to comment that these defaults were lifted directly from CT so that in the future we don't think of them as having been specifically designed?

Reply:
That's a good point. I think I'll exclude `Backoff` and `MaxBackoff` and just rely on the `Finalize` in CT if the user doesn't specify. The real improvement is the unlimited attempts.